src/Network/Network/NetworkWatcher/ConnectionMonitor/SetAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -89,48 +143,191 @@ public class SetAzureNetworkWatcherConnectionMonitorCommand : ConnectionMonitorB
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The ID of the connection monitor source.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The ID of the connection monitor source.",
+            ParameterSetName = "SetByConnectionMonitorV1")]
         [ValidateNotNullOrEmpty]
         public string SourceResourceId { get; set; }
 
         [Parameter(
             Mandatory = false,
-            HelpMessage = "Monitoring interval in seconds. Default value is 60 seconds.")]
+            HelpMessage = "Monitoring interval in seconds. Default value is 60 seconds.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Monitoring interval in seconds. Default value is 60 seconds.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Monitoring interval in seconds. Default value is 60 seconds.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Monitoring interval in seconds. Default value is 60 seconds.",
+            ParameterSetName = "SetByConnectionMonitorV1")]
         [ValidateNotNullOrEmpty]
         public int? MonitoringIntervalInSeconds { get; set; }
 
         [Parameter(
             Mandatory = false,
-            HelpMessage = "Source port.")]
+            HelpMessage = "Source port.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Source port.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Source port.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Source port.",
+            ParameterSetName = "SetByConnectionMonitorV1")]
         [ValidateNotNull]
         [ValidateRange(1, int.MaxValue)]
         public int SourcePort { get; set; }
 
         [Parameter(
              Mandatory = false,
-             HelpMessage = "The ID of the connection monitor destination.")]
+             HelpMessage = "The ID of the connection monitor destination.",
+             ParameterSetName = "SetByResource")]
+        [Parameter(
+             Mandatory = false,
+             HelpMessage = "The ID of the connection monitor destination.",
+             ParameterSetName = "SetByName")]
+        [Parameter(
+             Mandatory = false,
+             HelpMessage = "The ID of the connection monitor destination.",
+             ParameterSetName = "SetByLocation")]
+        [Parameter(
+             Mandatory = false,
+             HelpMessage = "The ID of the connection monitor destination.",
+             ParameterSetName = "SetByConnectionMonitorV1")]
```

Comment:
I don't think the parameter sets quite work.

I think you need

SetByResouirceV1
SetByResourceV2
SetByNameV1
SetByNameV2
SetByLocationV1
SetByLocationV2

In order for parameter sets to be valid, you need to have a required parameter that is unique to that parameter set.

Reply:
Actually, it works like what it is in the code. Any other arrangement gives error in powershell not able to resolve the parameter set.

Let me explain how this cmdlet shall be designed.

We have three sets:
SetByResource
SetByName
SetByLocation

the user can choose only one from the above set. With that, he can also choose any one from the following set:
SetByConnectionMonitorV1
SetByConnectionMonitorV2

so we have a total of 3 times 2 =  6 different combination of parametr set.

The one you see in the code is working . When i try to optimize , it gives error not able to resolve

Reply:
done